modules/tinymce/src/core/main/ts/fmt/Preview.ts

```diff
@@ -185,9 +186,9 @@ const parseSelector = function (selector: string) {
 
 const getCssText = function (editor: Editor, format) {
   let name, previewFrag;
-  let previewCss = '', parentFontSize, previewStyles;
+  let previewCss = '', parentFontSize;
 
-  previewStyles = editor.settings.preview_styles;
+  let previewStyles: string | boolean | undefined = Settings.getPreviewStyles(editor);
```

Comment:
These types should be on the return value of `getPreviewStyles`, not here. Plus you haven't fixed the original issue I mentioned here... For starters, all the default values below should be in the settings so we have consistent defaults wherever this is used. If you do that, then this function can just always return a string and you don't have to do the hacky `as string` cast below and also removes the mutation currently being done.

Reply:
The problem is that it's using them in different ways. If it returns false we immediately return an empty string, but if it returns undefined we later uses a default value.
The default value might be worthwhile moving into the setting, but it'd still need to return a `string | boolean` to type-fit later uses of it in this file where it sets this value to a different one, which may be 'false'.

Reply:
You don't though, as you can just replace the current `false` check with something like this:
```
if (previewStyles === '') {
  return '';
}
```

Reply:
I'm referring to the later part
`  previewStyles = format.preview;
    if (previewStyles === false) {
      return '';
    }`
which again makes false a valid value.

Reply:
That should use a different variable then, as it's mutating the original.

Reply:
Another option that keeps the mutation is to do something like this:
```
  const previewOpt = Obj.get(format, 'preview');
  if (previewOpt.is(false)) {
    return '';
  } else {
    previewStyles = previewOpt.getOr(previewStyles);
  }
```